src/coffee/helpers/task-queue.coffee

```diff
@@ -49,13 +50,19 @@ class TaskQueue
     @_activeCount += 1
 
     task.fn()
-    .then (res) ->
-      task.resolve res
-    .catch (error) ->
-      task.reject error
+    .then (res) =>
+      # postpone resolving the task
+      @_completedTasks.push(task.resolve.bind(this, res))
+    .catch (error) =>
+      # postpone rejecting the task
+      @_completedTasks.push(task.reject.bind(this, error))
```

Comment:
I think this depends solely on the use case we want to follow, if we want to collate all error then this looks right, if we want to fail on error then we go with the logic you suggested...